ftp.go

```diff
@@ -108,6 +113,32 @@ func DialTimeout(addr string, timeout time.Duration) (*ServerConn, error) {
 	return c, nil
 }
 
+// StartTLS Uses the AUTH TLS command to setup a TLS tunnel on the connection.
+func (c *ServerConn) StartTLS(config *tls.Config) error {
```

Comment:
What about creating a `func DialTLS(addr string, timeout time.Duration, config *tls.Config) (*ServerConn, error)` method?

Reply:
Is also a possibility, but would be less flexible than the current approach in the way that the application can handle TLS. The advantage of STARTTLS is that you can upgrade an existing connection and also handle the case that "no upgrade is possible" in the application. The application could decide that it wants TLS if possible but still continue to function if that request is denied (most probably depending on some user setting).

If you don't want that flexibility I can change it to provide a ` DialTLS`  function instead. Another possibility would be to add a `DialTLS`  but still keep the existing `StartTLS`  function by making `DialTLS` use both existing functions.